x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextField.java

```diff
@@ -0,0 +1,328 @@
+    implements
+        ToXContentObject {
+
+    static final ParseField TEXT_FIELD = new ParseField("text");
+    static final ParseField INFERENCE_FIELD = new ParseField("inference");
+    static final ParseField INFERENCE_ID_FIELD = new ParseField("inference_id");
+    static final ParseField CHUNKS_FIELD = new ParseField("chunks");
+    static final ParseField CHUNKED_EMBEDDINGS_FIELD = new ParseField("embeddings");
+    static final ParseField CHUNKED_TEXT_FIELD = new ParseField("text");
+    static final ParseField MODEL_SETTINGS_FIELD = new ParseField("model_settings");
+    static final ParseField TASK_TYPE_FIELD = new ParseField("task_type");
+    static final ParseField DIMENSIONS_FIELD = new ParseField("dimensions");
+    static final ParseField SIMILARITY_FIELD = new ParseField("similarity");
+
+    public record InferenceResult(String inferenceId, ModelSettings modelSettings, List<Chunk> chunks) {}
+
+    public record Chunk(String text, BytesReference rawEmbeddings) {}
+
+    public record ModelSettings(TaskType taskType, Integer dimensions, SimilarityMeasure similarity) implements ToXContentObject {
+        public ModelSettings(Model model) {
+            this(model.getTaskType(), model.getServiceSettings().dimensions(), model.getServiceSettings().similarity());
+        }
+
+        public ModelSettings(TaskType taskType, Integer dimensions, SimilarityMeasure similarity) {
+            this.taskType = Objects.requireNonNull(taskType, "task type must not be null");
+            this.dimensions = dimensions;
+            this.similarity = similarity;
+            validate();
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject();
+            builder.field(TASK_TYPE_FIELD.getPreferredName(), taskType.toString());
+            if (dimensions != null) {
+                builder.field(DIMENSIONS_FIELD.getPreferredName(), dimensions);
+            }
+            if (similarity != null) {
+                builder.field(SIMILARITY_FIELD.getPreferredName(), similarity);
+            }
+            return builder.endObject();
+        }
+
+        private void validate() {
+            switch (taskType) {
+                case TEXT_EMBEDDING:
+                    if (dimensions == null) {
+                        throw new IllegalArgumentException(
+                            "required [" + DIMENSIONS_FIELD + "] field is missing for task_type [" + taskType.name() + "]"
+                        );
+                    }
+                    if (similarity == null) {
+                        throw new IllegalArgumentException(
+                            "required [" + SIMILARITY_FIELD + "] field is missing for task_type [" + taskType.name() + "]"
+                        );
+                    }
+                    break;
+                case SPARSE_EMBEDDING:
+                    break;
```

Comment:
> I guess that would make us more forward compatible in case this is used in a mixed cluster env, but I could be wrong. Should we be more strict for non supported params?

But then the old mappings wouldn't even be using those values and consequently, we end up with things indexed very differently in a mixed cluster environment? 

I think we should be strict here. We shouldn't allow parameters that are invalid.